bodhi/server/templates/update.html

```diff
@@ -161,7 +161,13 @@
         // note: these are actually single-item arrays, though this doesn't
         // seem to break anything
         var arch = result.data.arch;
+        if (arch === undefined){
+            arch = 'noarch';
+        }
         var item = result.data.item;
+        if (item === undefined){
+            item = 'AtomicCi';
```

Comment:
If I'm reading the code correctly, this will not work, since `item` is supposed to be NVR, so that if an update contains multiple NVRs, we compute the latest one for each of them. If you merge all item-less results under `AtomicCi`, you get just a single result for multi-build updates. Has this been tested? I might be wrong.

What you really want to do here is to use `original_spec_nvr` if it exists, I think. Ideally just when the testcase matches. Btw, is there a good reason that centosci tasks don't fill out `item`? At least for those testcases where it makes sense (like this one)?

Reply:
I have tested it with an update with a single build indeed, I'll see if I can adjust this to be ``original_spec_nvr`` as well.

As for why there is no ``item`` I have no idea, because @tflink designed it this way? What is ``item`` supposed to be?

Reply:
`item` is supposed to be the common identifier of the thing that got tested. So for `type=koji_build`, it's NVR. For `type=bodhi_update`, it's update ID. See the usual `item+type` combinations here:
https://qa.fedoraproject.org/docs/libtaskotron/latest/library.html#module-libtaskotron.main

I don't exactly know what `org.centos.prod.ci.pipeline.complete` tests, can you explain? I don't know if there's a `type` that would match this (we can introduce a new one), but so far the `item` seems to look like Koji NVR (i.e. in [results/17433016](https://taskotron.fedoraproject.org/resultsdb/results/17433016) it seems that `cockpit-151-2.f26` is the thing that got tested). Does that sound correct?

If @tflink designed this, it might be a good idea to wait for him to provide details in here.

Reply:
If it is the thing that got tested, then I guess it should be the git hash of the commit that got tested. Basically the Atomic CI pipeline triggers for every commit made in dist-git. The  ``org.centos.prod.ci.pipeline.complete`` is then sent at the end of every run, successful or not, and it is then stored in resultsdb (as well as intermediary results).

But the git hash isn't really human friendly either :/
I'll try to put the ``original_spec_nvr`` instead so the UI is prettier

Reply:
Well then, we already have exactly this `item+type` combination defined, an example:
```
type=dist_git_commit
item=modules/dhcp#5c6331592b6c92bf7c52a2ac735f92cb8fc11304
```
That's the right thing to submit to resultsdb, because then it will be easy to find results for dist git changes regardless of a testcase.

You see, the item+type standard is a way to make different testcases play along together nicely, to have a common standard. If we don't have that, each testcase will have its own approach for specifying what got tested, and the test consumers will be extremely difficult to write, having to know special cases for each testcase (you can see it here). So I'd very much recommend to use the aforementioned approach here. If all distgit tests use the same item+type definition, then we can query all the results, regardless whether they are centosci, taskotron or anything else.

One further complication here is that bodhi seems a bit unfit to display distgit commit results. Bodhi shows koji builds grouped to updates. Do I understand correctly that you want to show the distgit commit result for a specfile version that corresponds to the koji build NVR? And what happens if somebody pushed several commits without changing specfile version, you just take the latest one (which might or might not be correct)? Sure, this can be done this way, but wouldn't it make more sense to run the centosci test suite on actual koji builds (like we do with rpmlint etc), display those in bodhi, and make a different frontend for displaying distgit commit results, really after each commit (I'm assuming our resultsdb frontend is not sufficient)?

Reply:
I managed to put the ``original_spec_nvr`` as well as the commit hash in there instead, was pretty straight forward at the end :)

This is going to be the good middle group I believe, but I'm willing to revisit if people feel otherwise.